### DISCUSSION TITLE
Logik in Homematic oder Home Assistant?

### DISCUSSION THREAD

**Discussion by ThomDietrich** [SukramJ/aiohomematic#475] [Q&A]
Hallo,
ich denke mir das kann auch für andere interessant sein.

In vielen der aktuell aufkommenden Diskussionen geht es um die detaillierte Unterstützung von Systemvariablen, Programme, Gruppen und Verknüpfungen, etc. in diesem Projekt - und damit eine Kontrollmöglichkeit in der Home Assistant Integration. Ich finde es durchaus erstrebenswert und toll den vollen Funktionsumfang von HM in HA abzubilden 👏 Kudos den Entwicklern hierzu.

Eine grundlegende Frage habe ich dazu allerdings doch: **Wozu nutzt ihr diese Funktionen?** Gibt es rationale Gründe HM und HA parallel für Logik zu nutzen?

Als Gegenbeispiel: Ich nutze meine Homematic Basis (CCU/Raspberrymatic) als reine Geräteschnittstelle (parallel zu meiner ZigBee Basis, der Alexa Integration oder eine MQTT Broker). Sämtliche Logik, inkl. Gruppen, Verknüpfungen und Zeitprogramme, implementiere ich in Home Assistant, technologieübergreifend. HA bietet eine modernere Oberfläche, mächtige Automatisierungsmöglichkeiten und keine merklichen EInbußen bei Latenz oder Zuverlässigkeit. Verknüpfungen oder Regeln in HM würden hier einen Stilbruch bedeuten und sind daher in meinen 7 Jahren HM Erfahrung nach und nach gewichen.

**Comment by SukramJ** [comment]
Meine Umgebung besteht im wesentlichen aus HmIP (Heizung, Lüftung, Sicherheit) und Zigbee (Licht, Sensoren)
Um die Zuverlässigkeit zu erhöhen (oder nicht zu verringern) bilde ich die grundlegenden Funktionen in den Zentralen direkt ab.

Bei HM verwende ich primär Direktverknüpfungen (sofern möglich), Programme (u.a. Sammeln von Sensoren für Hüllschutz, Vollschutz, Alarmsteuerung, ..) und im Zweifel auch mal ein Script. Die Programme schreiben Stati in Systemvariablen.
Einige Steuerungsfunktionen sind über Wertelisten abgebildet.
Alle Geräte und Kanäle haben vernünftige Namen, sind Räumen zugewiesen und haben teilweise Gewerke.

Bei Zigbee nutze Gruppen aber keine Szenen.
Übergreifende Automatisierungen, Scripte und alle Szenen werden in HA abgebildet.

Da ich HM schon viel länger als HA nutze sind einige Sachen natürlich gewachsen, aber Sachen die robust laufen muss ich auch nicht ändern.
Die Möglichkeiten in HA waren früher deutlich limitierter, bzw. weniger intuitiv. Da hat sich in der letzten Zeit aber einiges getan.

**Reply by ThomDietrich** [reply]
Ich verstehe. Du bestätigst meine Vermutung, dass es primär um den Support historisch gewachsener Installationen geht. Das ist völlig in Ordnung. Hier greift das Prinzip "Never change a running system" - auch wenn ich das persönlich nicht so halten würde. Altlast bringt Last.

> Die Möglichkeiten in HA waren früher deutlich limitierter, bzw. weniger intuitiv. Da hat sich in der letzten Zeit aber einiges getan.

Home Assistant habe ich auch erst seit circa zwei Jahren im Einsatz. Vorher war openHAB bereits einige Jahre weit vor den Möglichkeiten von HM. Meine letzte Direktverknüpfung habe ich vor circa 4 Jahren gelöscht

**Reply by SukramJ** [reply]
> Ich verstehe. Du bestätigst meine Vermutung, dass es primär um den Support historisch gewachsener Installationen geht. Das ist völlig in Ordnung.

Darum geht es mir absolut nicht. Ich würde viele Sache aktuell genau so wieder machen.
Warum sollte ich in meinen Augen sinnvolle Funktionen nicht nutzen wollen.
Ich verstehe nicht woran Du deine Vermutung knüpfst.

**Reply by chrste90** [reply]
Und gerade Direktverknüpfungen bei HM haben ja noch den Vorteil, dass diese auch ohne laufende Zentrale funktionieren. Deswegen würde ich die auch auf jeden Fall beibehalten, egal wo die sonstige Logik umgesetzt wurde.

**Reply by ThomDietrich** [reply]
@chrste90 Das ist ein guter Punkt. Wenn bei mir zuhause der HA Host ein Update installiert scheint im Bad kein Licht. Das ist eine Schwachstelle welche auch Home Assistant so schnell nicht lösen wird (Redundante Installationen sind alles andere als trivial in dem Kontext).

Allerdings... Mein Bad-Taster ist von Homematic, die Spiegelbeleuchtung wird von einem Sonoff Relais geschaltet, in der Deckenlampe steckt eine Tradfri Leuchte - Der Wunsch nach Direktverknüpfungen hätte mir hierbei mehr Restriktionen als Vorteile gebracht :)

**Comment by danielperna84** [comment]
Ich habe die Logik (und alles drum herum) - von Direktverknüpfungen abgesehen - ausschließlich in HA. Das kommt daher, dass ich anfangs _nur_ Homegear genutzt habe als ich bei HA eingestiegen bin. Und bei Homegear gibt's ja keine Logik. Genau deshalb habe ich überhaupt mit HA angefangen (um eine Logikschicht zu haben), denn ioBroker und OpenHAB haben mir nicht zugesagt. Und von FHEM wollte ich ganz dringend weg.

**Reply by ThomDietrich** [reply]
Wir scheinen da recht ähnliche Vergangenheiten zu haben. Ich habe vor vielen Jahren mit FHEM begonnen, bin schnell zu Homematic als Platform gewechselt. Habe dieses dann durch openHAB ergänzt und habe meine CCU2 durch Homegear ersetzt. Die letzten paar Jahre nutze ich nun Home Assistant mit RaspberryMatic.

Die Logik war bei mir sofort in openHAB/Home Assistant gewandert. Einige Direktverknüpfungen hatte ich bis vor zwei-drei Jahren noch in Homegear.

Das einzige Argument für Direktverknüpfungen oder gar HM Programme ist in meinen Augen Performance. Eine Taster -> Lichtrelais Kopplung war in openHAB mit zu hoher Latenz verbunden, was eine Direktverknüpfung erzwungen hat. Seit HA sind diese Probleme vom Tisch. 

Für Logik in HA spricht für mich ganz klar die systemübergreifende Automation, inklusive der bedingten Ausführung, welche mit Direktverknüpfungen nicht möglich ist.

**Reply by riesi77** [reply]
Ich habe zwar nicht wirklich viel Logik in Homematic - aber ich bin froh über Alternativen. Durch die in der letzten Version geschaffene Möglichkeit, Homematic Programme direkt als Buttons in HA einzubinden ist beides möglich. Ich denke es wird Benutzer gibt, die komplexe, funktionsfähige Programme in Homematic haben und diese über HA nur aufrufen wollen. Und es wird welche geben, die die saubere "alles ist in HA" Lösung bevorzugen. Wobei ja die Direktverknüpfungen, die ich auch als eine Form der Logik sehe, ja nach wie vor nur direkt in Homematic zu finden sind.
Insofern ist das für mich kein entweder/oder, sondern bis zu einem gewissen Grad auch Geschmachssache.

**Reply by ThomDietrich** [reply]
Und ich denke genau da will ich niemandem widersprechen :) Ich halte das ebenso für einen völlig legitimen Grund, insbesondere in einer Übergangs- und Testphase.

Direktverknüpfungen gibt es konzeptionell in HA nicht. Das ergibt aber auch Sinn, da sie letztlich auch nur der einfachste Grad einer Automation sind. So sieht das bei mir aus. Das einfache Beispiel berücksichtigt noch keine `long_press` events oder `conditions`:

```yaml
  - alias: Regel Taster Küche schaltet Küchenbeleuchtung
    trigger:
      - platform: event
        event_type: homematic.keypress
        event_data:
          address: REQ0574736
          type: press_short
    action:
      service: {{ 'light.turn_on' if trigger.event.data.subtype == 1 else 'light.turn_off' }}
      entity_id: light.kueche_deckenlicht
```

**Reply by emufan** [reply]
... was aber nicht geht, wenn CCU oder HA-Server nicht erreichbar sind, usw. Daher bleiben z.B. für solch einfache Sachen definitiv und wo immer möglich die Direktverknüpfungen. Programme in der CCU brauche ich dagegen gar nicht.

**Comment by kwithus** [comment]
Moin, ich komme auch ursprünglich mal von FHEM, bin dann zu Homematic ‚umgezogen‘ dort zuletzt viel mit NodeRed gemacht.
Seit ich auf HomeAssistant gestoßen bin, habe ich alle NodeRedFlows mittlerweile durch HA Automations und Scripts ersetzt.
Ebenso fast alle verbliebenen CCU Programme.
Die CCU nutze ich zum einen, dort wo für mich machbar, für Direktverknüpfungen und für ein paar ‚Backup_Funktionen‘ so habe ich zum Beispiel in den Rollladenaktoren überall Wochenprogramme hinterlegt, die (mit ausreichend zeitlichen Abstand zu den HA-Einstellungen) die Rollläden hoch/runterfahren. Da diese direkt in den Brolls liegen, funktionieren sie eben auch, wenn CCU/HA einmal komplett ausfallen sollten (und ich aus welchen Gründen auch immer nicht eingreifen kann)
In HA kann ich zudem einstellen, ob die Wochenprogramme insgesamt ausgeschaltet werden, abhängig von der Einstellung für Rolllden-Automatik in HA aktiv oder deaktiv sind, oder immer aktiv sind.
Das nur als Beispiel für eine solche Backup-Funktionalität.
Aber im Wesentlichen möchte ich alles in HA programmieren.

**Reply by ThomDietrich** [reply]
Hey @kwithus,
guter Punkt mit den "Backup-Funktionen". Gerade in Kombination mit deinen Rolladen halte ich das für eine sehr sinnvolle Idee.

**Comment by addy38** [comment]
Hallo, für mich ist derzeit der Grund an der Logik in HM festzuhalten das deutsche HM Forum und CUxD.
Im HM Forum ist fasst jeder Anwendungsfall irgendwie zu finden. Somit komme ich oftmals auch ohne Skripterfahrung mit Paste+Copy ans Ziel. Im Gegensatz dazu ist HA Support leider nur über das englischsprachige Forum zu finden und HM dort leider nur sehr gering vertreten.

**Reply by emufan** [reply]
Das sehe ich wiederum gegenteilig, witzig, wie unterschiedlich die Ansichten sind. 

Das HA-Forum, Discord, die HA-Doku, usw. das ist für mich einfach fantastisch umfassend, die Leute mehrheitlich hilfreich, nett und schnell.

Dagegen gehört das HM-Forum zu der Gruppe, der Foren, die alles zeigt, wie Foren mMn nicht sein sollten. Die Antworten sind zu 95% entweder "probier doch selber" oder "frag doch den Hersteller" oder die 20 Jahre-Forum-Oldies antworten sich nur untereinander zu Lasten der anderen.

Und bevor ich mich überhaupt in die Untiefen der HM-Scriptsprache per Copy&Paste ranmache (was ich technisch sogar könnte), klicke ich mir die gleiche Logik in HA einfach mal schnell zusammen oder nutze die dortige, einfache Templateprogrammierung zur Hilfe.

**Comment by kwithus** [comment]
ich sehe es wie emufan, ich verstehe aber, dass Englisch als Forensprache manchen abschreckt.
Aber das ‚Programmieren‘ ist in HA viel einfacher 

Von meinem iPhone gesendet

**Comment by SukramJ** [comment]
Mittlerweile gibt es ja so einige Rückmeldungen, aber was konkret in HA genutzt wird schreibt keiner(?).
Ich beziehe mich da auf die Nutzung von Templates, Scripte, Scenen und Automatisierungen.
Insbesondere finde ich interessant ob ihr Automatisierung über UI (only) oder über YAML verwendet.
Wie sieht es eigentlich mit der Nutzung von NodeRed und AppDaemon aus.

**Reply by kwithus** [reply]
Ich habe angefangen mit HA und NodeRed, da i h NodeRed auch in Homematic intensiv nutzte. Das war für die Umstellung am einfachsten.
Mittlerweile habe ich alles auf HA Automations umgestellt, grundsätzlich alles in der UI, einige Teile innerhalb von Automations auch yaml.
Szenen nutze ich so gut wie gar nicht, aber scripte, die von Automations aufgerufen werden.
Blueprints nur vereinzelt, templates und selbst definierte Sensoren, Variables, etc.
Neben Core nutze ich einiges an Integration und custom components aus hacs. Eine wichtige Integration ist HomeKit, daneben z.B. Möhlenhoff Alpha2, Overkiz und andere.
Die Darstellung erfolgt im Dashboard und parallel in HomeKit.

**Reply by emufan** [reply]
Templates im Zusammenhang mit HM: Meist nur Werterechnungen Valve-States für Anzeigen in Graphen, die mit binary-sensor nichts anfangen können. Oder spezielle Dinge wie Stromverbrauch Rollläden aufgrund der Status. Oder Hilfsentitäten, dass ein Rollladen nur zu, wenn ganz zu und nicht nur wenn >0. Oder Trockner-Betriebesstatus nebst Pushmeldungen mit PSM. Oder Filter auf Helligkeits- und Temperatursensoren. Ebenso Extremwerte. Etc. Etc.

Nodered & Co. No. Alles HA. Wüsste aktuell auch nicht, wofür ich das brauchen sollte.

Szenen. Nichts. Habe ich lieber in den Automatisierungen und Skripten.

Automatisierungen. UI/YAML. Sehe da den Unterschied nicht. Ist ja beides das gleiche, von ganz wenigen Dingen, die noch nicht im UI sind, vielleicht abgesehen. Starte meist im UI. Habe ich irgendwas schon mal oder möchte Pfade Duplizieren innerhalb einer Automatisierung, wechsle ich in den YAML-Editor und kopiere da vor mich hin. Dort, wo ich keine Typos haben will, wechsle ich wieder zurück in den Plain-UI-Modus und selektiere Entitäten, usw. Ein stetiges hin und her per (wie heißt das Ding?) Drei-Punkte-Menü. Interesse: Warum die Frage nach Automatisierung UI (only) oder über YAML?

Automatisierungen: En masse: Rolladensteuerung. Lichtstatus. Gerätestatus bei Ab- und Abwesenheiten. Lampen aus, wenn lange an. Geräte aus, wenn x oder y. Benachrichtigungen und Alerts Status Fenster und Türen. Steuerung für Notfälle (Rauchmelder, ...). Bewegungsmeldersteuerung mit mehr Feinheiten. Zu weitere Alerts und Benachrichtigungen. Uvm.

**Reply by kwithus** [reply]
Dieser schnelle Wechsel zwischen UI und YAML, z.B. um umfangreiche Automationen per copy Paste anzulegen oder auch im Dashboard komplexere Card-Definitionen zu kopieren gefällt mir auch bei HA extrem gut. 
wobei ich mir bei verschachtelten Automationen in UI manchmal eine etwas bessere Übersicht wünschen würde (z.B. bei Auswahlaktionen / und/oder Bedingungen Wenn-Dann Aktionen könnten die jeweils zugehörigen Rahmen farblich unterschiedlich sein)
Aber insgesamt ist die Programmierung wesentlich übersichtlicher und einfacher als in HM

**Reply by SukramJ** [reply]
> Warum die Frage nach Automatisierung UI (only) oder über YAML?

Ich hatte bei Issues manchmal das Gefühl das recht viel YAML ohne UI verwendet wird. Typische Frage: Woher bekomme ich die `device_id`?

**Reply by emufan** [reply]
Ah. Verstehe.

BTW: Geräte, `device_id`s nutzte ich persönlich z.B. eigentlich zu 0,0. Nur `entity_id`s. Generell und nicht HA-spezifisch. So kann ich bei Geräteaustausch einfach die alte `entity_id` wiedernehmen und muss dadurch nichts an Templates, Skripten, Automatisierungen, usw. ändern.

**Reply by chammp** [reply]
Schon ein bisschen alt hier, aber ich dachte, ich gebe mal meinen Sönf dazu 😄 

Ich nutze von HA eigentlich alles, was geht (und wozu ich eine Verwendung in Form von Sensor oder Aktor habe). Templates, Skripts, Skripts als Funktionen, Automationen (haha), Template Sensoren, - Schalter, - Locks. 

Auf der CCU laufen Direktverknüpfungen für essentielle Sachen, z.B. Schalter -> Rollos, Schalter -> Licht etc., in HA liegt dann der Convenience-Layer (z.B. Schalter -> Licht in der passenden Farbtemperatur). Davon ab bin ich systemübergreifend unterwegs und habe auch einiges Zigbee-Gesocks hier im Einsatz, das geht dann _immer_ über HA.
Außerdem nutze ich auf der CCU Skripte/kleine Programme, um in ausgewählten Situationen Dinge dort zu beeinflussen, die über HA nicht gehen (bzw. die ich nicht rausgefunden habe). So nutze ich z.B. die virtuellen Kanäle 5 und 6 bei den Rollos allgemein für Max/Min-Behanghöhe, die von verschiedenen Faktoren beeinflusst wird. Damit man an den Schaltern allerdings fast (siehe gleich) alle Einschränkungen durch Programme/Automatisierungen überschreiben kann, setzen die DVs auf den Schaltern die Min/Max-Grenzen wieder zurück. So, und jetzt habe ich aus aktuellem Anlass einen Frostschutz gebaut, der die Min-Behangöhe steuert. Das geht über die initial deaktivierten vch5/6-Entitäten, nur wenn jetzt jemand am Schalter zumacht, dann möchte ich, dass in _diesem_ Fall (und nur da) der Frostschutz nicht überschrieben werden kann. Deshalb wird dann die DV umparametrisiert und der Begrenzungslevel geändert. Das läuft auf der CCU, getriggert durch Übergang einer Systemvariablen wiederum ausgelöst durch einen HA-Schalter.

Achso, und auf HA selber mache ich eigentlich alles was geht direkt in YAML, weil ich in Software mache und ich das natürlicher finde. (Also, Quelltext jeder Art ist für mich natürlich, in YAML zu "programmieren" fühlt sich immernoch eklig und falsch an - aber trotzdem noch besser als über die grausliche UI, erst recht, wenn es etwas komplizierter wird.) Ich nutze auch sehr sehr (SEHR!) intensiv Packages. Ich frage mich, wie man ohne Packages größere Installationen noch verstehen und verwalten können will, aber vielleicht bin ich auch nur zu doof 😉

**Reply by ThomDietrich** [reply]
@chammp danke für deinen Sönf :) Bei mir passiert Null-Komma-Nix in der CCU. Direktverknüpfungen waren mir früher oder später im Weg (bzw. plötzlich war Logik versteckt in anderen Systemen außerhalb von HA) und zusätzliche Funktionen sah ich nicht. Interessant dein Anwendungsfall mit den Rollos. Dazu gibt es keine Möglichkeit in HA? (Nur interessehalber, ich möchte den Weg nicht kritisieren).

> Ich nutze auch sehr sehr (SEHR!) intensiv Packages.

Hier sprichst du mir aus der Seele! Wieso eine Packages Vorbereitung nicht Teil der standardmäßigen Config eines neuen HA Setups ist ist mir völig unklar (Vielleicht sollte da einfach mal wer ein PR eröffnen 😇

**Reply by chammp** [reply]
Das mit den Rollos kannst du auch mit HA-Hausmitteln machen. Die vch5/6-Entitäten jeweils aktivieren, schlau benennen (z.B. `cover.blabla_max` und `cover.blabla_min`), in der CCU entsprechend einstellen (Kanal 5 ist dann UND und 6 ODER - Gedanke dahinter ist, dass aus Notfallgründen lieber offen als zu sein soll, wenn es sich widersprechende Settings gibt) und das müsste es so ziemlich gewesen sein (müsste weil ausprobiert hab ich's nicht - s.o.). Die Min/Max-Einschränkungen macht dann der Aktor für dich.

_Ich_ möchte aber alles essentielle ohne HA/CCU laufen haben, weshalb es bei mir DVs sein müssen. Ich hatte tatsächlich vor kurzem das Problem, dass HA (inkl. CCU - RaspberryMatic als HA-Addon) nach einem OS-Update überhaupt nicht mehr da war. Weil ich auch vergessen hatte, wie der SSH-Zugang funktionierte, waren wir 2h lang in einem total doofen Haus, naja, mit Ausnahme aller essentiellen Grundfunktionen, die eben über DVs liefen 😅 So war es immerhin für niemanden allzu nervig. Hier hat sich mein gewisses Level an Paranoia ausgezahlt 😜 

Die Taster an der Wand mit ihrer Min-Max-Überschreiben-Logik-Es-Sei-Denn-Es-Ist-Frostschutz-An kannst du natürlich auch in HA abbilden. Auslösungen durch Taster sowieso, weil du ja in HA auf die Taster-Events reagieren müsstest, und wenn du es auch über ein UI-Control machen können willst, legst du dir eine [template cover](https://www.home-assistant.io/integrations/cover.template/) Entity an, nennst das Kind `cover.blabla_ui_control`, leitest als Positions- und Status-Templates die eigentliche Entität `cover.blabla` durch, und in den Service-Scripts für Hoch, Runter und Position setzen resettest du nach Gutdünken und Bedarf die zugehörigen `_min`- und `_max`-Entities und leitest danach den gewünschten Befehl an `cover.blabla` weiter. Automationen, die sich innerhalb deiner Grenzen bewegen sollen (z.B. Beschattung, die überschrieben wird durch einen Drehgriffsensor) regeln dann direkt `cover.blabla`, und wenn du eine Automation hast, die besonders wichtig ist (getriggert durch Rauchwarnmelder oder ähnliches), kann diese ja stattdessen gleich `cover.blabla_ui_control` steuern.

Achso, der Grund für den Ausfall war übrigens ein typisches Layer8-Problem, also ICH habe die yaml-Konfig kaputtgebastelt und das vergessen und viel später gedankenverloren auf den HA-OS-Update-Knopf gedrückt. _Eigentlich_ hätte HA dann in einen recovery-Modus wechseln sollen, statt ganz abzurauchen, hat es aber nicht, und deshalb arbeite ich gerade an einem PR, der genau das macht und mir den Popo wenn nicht gerettet dann doch wenigstens die Ausfallzeit reduziert hätte. Hoffentlich geht der durch, wenn er fertig ist 🤞 

> Wie sieht es eigentlich mit der Nutzung von NodeRed und AppDaemon aus.

NodeRed hatte ich mal, lohnte sich aber für mich nicht, also ist es rausgeflogen. AppDeamon überlege ich immer mal wieder ernsthaft, ob ich komplexere Späße nicht damit regeln will, aber bisher hab ich's noch immer mit HA-Bordmitteln in wartbar hinbekommen...

**Reply by benkly1980** [reply]
Ich habe aktuell das Problem das ständig Geräte in HA nicht verfügbar sind. Manchmal reicht es dann die Integration neu zu laden, manchmal muss ich über die HM UI den Aktor einmal hin und her schalten, dann ist es in HA wieder aktiv.
Hab bisher nicht heraus gefunden wie ich das in den Griff bekomme.

**Reply by SukramJ** [reply]
@benkly1980 Hat diese Diskussion irgendwas mit deinem Problem zu tun? 
Ich würde an deiner Stelle mal die mal die [Doku](https://github.com/danielperna84/custom_homematic#homematicip_localforce_device_availability) lesen.

**Comment by kwithus** [comment]
Ich denke, das ist ein Anfangs/Anfänger-Problem. Ich musste auch erst den Umgang mit der UI lernen

Von meinem iPad gesendet

**Comment by ivo7B4** [comment]
Ich habe ziemlich zeitgleich mit HomeAssistant und CCU3 begonnen. Bei der CCU3 habe ich mehrheitlich wired Komponenten. Ich habe mich in beide Systeme und die entsprechenden Automatisierungslogiken daher gleichzeitig eingearbeitet, wegen des HmIP-WRCD (welchen ich ebenfalls verwende) zusätzlich auch in die Script-Logik der CCU3.

Für eine möglichst hohe Zuverlässigkeit und auch für eine Abstraktion (einfache Logiken so lokal wie möglich), habe ich die Direktverknüpfungen alle in der CCU3 erstellt. Ein Lichtschalter z.B. funktioniert damit auch dann, wenn die CCU3 oder HomeAssistant nicht läuft.

Da es die Wetterstation und den Regensensor (zumindest soweit ich weiss) bisher nur als Funkvariante gibt, mir dort aber für den Markisenschutz eine sehr hohe Zuverlässigkeit wichtig ist, habe ich dies als eine Automatisierung in der CCU3 gelöst (damit ich beide Geräte in einer Automatisierung als trigger für's Hochfahren der Markise verwenden kann). Dies als DVs habe ich mir überlegt, sehe hier jedoch keinen Vorteil für die Zuverlässigkeit, da die CCU3 bei DVs zwischen Funk- und wired Komponenten als Gateway ebenfalls laufen muss. Hier würde ich klar nicht über HomeAssistant gehen. Ich bin mir auch nicht sicher, ob nur die "average Windwerte" oder auch kurzfristige höhere Windgewschwindigkeiten von der Wetterstation an HomeAssistant übermittelt werden, welche für eine ausreichend rasche Reaktionsgeschwindigkeit erforderlich sind.

Was aber nicht so super zuverlässig sein muss oder komponentenübergreifend ist (z.B. Taster an einem HmIP IO Modul und damit toggle z.B. von Zigbee-Lampen), habe ich alles in HomeAssistant realisiert. 

Ich würde dies auch neu wieder so umsetzen.

**Reply by HiranChaudhuri** [reply]
> Hier würde ich klar nicht über HomeAssistant gehen. Ich bin mir auch nicht sicher, ob nur die "average Windwerte" oder auch kurzfristige höhere Windgewschwindigkeiten von der Wetterstation an HomeAssistant übermittelt werden, welche für eine ausreichend rasche Reaktionsgeschwindigkeit erforderlich sind.

Ich habe die DWD Integration aktiv, und sie liefert die Windgeschwindigkeit. Auch Windböen sind erhältlich - allerdings musste ich diese Entity in Homeassitant separat aktivieren.


> Was aber nicht so super zuverlässig sein muss oder komponentenübergreifend ist (z.B. Taster an einem HmIP IO Modul und damit toggle z.B. von Zigbee-Lampen), habe ich alles in HomeAssistant realisiert.
> 
> Ich würde dies auch neu wieder so umsetzen.

Dies sehe ich genauso. Wobei ein Wermutstropfen besteht: In meiner CCU2 kann ich die HMScripte nicht per Versionskontrolle sichern/wiederherstellen. Da ich keine CCU3 habe und in Homeassistant nicht eingearbeitet bin, habe ich noch keinen Vergleich.
